src/main/java/org/openlmis/integration/dhis2/repository/indicator/CceRepository.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
+
+package org.openlmis.integration.dhis2.repository.indicator;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class CceRepository {
+
+  static final String STATUS = "status";
+  static final String UTILIZATION = "utilization";
+
+  @PersistenceContext
+  EntityManager entityManager;
+
+  /**
+   * Retrieves CCE count for a given status.
+   */
+  public String findCceCountByStatus(@Param(STATUS) String status) {
```

Comment:
Why this and other 'indicator calculating' methods return Strings?

Reply:
That's the best way I've found to initialize BigDecimal from a query. Is there a better way to do it?

Reply:
The issue here is that you have a code which calculates something what is supposed to be a number, but returns String - that looks confusing.
I think what you could do is, given findCceCountByStatus returns a count, the fitting type is long, then return long. Later when your context is 'an indicator', convert the long to BigDecimal as that is the type of all indicators. 
Other code to calculate indicator values would follow the same pattern: 
1st layer is a query which reads something from DB and have a type most fitting for what it read (count = long, sum = maybe double, a sum of BigDecimals = BigDecimal); 
2nd layer you run the query to get a value and you give it a meaning that it's an indicator, so now you convert it to BigDecimal as this is indicator's type.

That way when someone sees CceRepository#findCceCountByStatus it will have a type they would expect.

Reply:
I changed return types for all of these methods